### PR TITLE
Make HashSet::insert return OccupiedEntry

### PR DESCRIPTION
**Pull request by SUPERCILEX**
See https://github.com/rust-lang/rust/pull/120077#discussion_r1475708665

### CODE REVIEW

**Comment by Amanieu** [comment]
Sorry for the late review, can you rebase to fix conflicts?

**Comment by SUPERCILEX** [comment]
Sure, done.

**Comment by decathorpe** [comment]
Reading this comment:
https://github.com/rust-lang/rust/pull/120077#discussion_r1475708665

And the changes (i.e. the signature of the `insert` function has changed), shouldn't this have been released as v0.15.0 instead of v0.14.4? Or is changing the return type of a function from `()` to non-`()` OK?

File: src/set.rs [hunk 1]
```diff
@@ -1003,7 +1003,7 @@ where
     ///         match singles.entry(ch) {
     ///             Vacant(single_entry) => {
     ///                 // We found a new character for the first time.
-    ///                 single_entry.insert()
+    ///                 single_entry.insert();
```
Comment (cuviper) [inline src/set.rs:1006]: This exemplifies how it was a breaking change to return something other than `()`.

**Comment by Amanieu** [comment]
I've yanked the release.

**Comment by SUPERCILEX** [comment]
Will there be a replacement release for the stdlib?

**Comment by Amanieu** [comment]
Yes, I'll prepare a new release next week. I've been busy at Rust Nation this week.

**Comment by SUPERCILEX** [comment]
Oh fun! Sounds good.

**Pull request merged**